kv/etcd/options.go

```diff
@@ -74,6 +74,12 @@ type Options interface {
 	// SetWatchChanInitTimeout sets the WatchChanInitTimeout
 	SetWatchChanInitTimeout(t time.Duration) Options
 
+	// WatchStartRevision is the revision that watch requests will start from.
+	WatchStartRevision() int64
+	// SetWatchStartRevision sets the revision that watch requests will start
+	// from.
+	SetWatchStartRevision(rev int64) Options
```

Comment:
Wonder if we should name this `SetWatchWithRevision` to be consistent with how this is named in etcd codebase.